File: examples/string-length/case-unicode.mjs
/**
 * Unicode Test Case
 *
 * Provides a string with emoji characters that differ
 * in .length vs [...str].length due to surrogate pairs.
 */
export function createCase() {
	return {
		case: {
			caseId: "unicode-emoji",
			caseClass: "unicode",
			name: "Unicode Emoji",
			version: "1.0.0",
			inputs: { text: "hello 🌍🎉" },
		},
		getInput: async () => ({ text: "hello 🌍🎉" }),
		getInputs: () => ({ text: "hello 🌍🎉" }),
	};
}

File: examples/string-length/case_unicode.py
"""Unicode Emoji Test Case."""


def create_case():
    return {
        "case": {
            "caseId": "unicode-emoji",
            "caseClass": "unicode",
            "name": "Unicode Emoji",
            "version": "1.0.0",
            "inputs": {"text": "hello \U0001f30d\U0001f389"},
        },
        "getInput": lambda: {"text": "hello \U0001f30d\U0001f389"},
        "getInputs": lambda: {"text": "hello \U0001f30d\U0001f389"},
    }
